operator/pkg/name/name.go

```diff
@@ -49,8 +49,15 @@ const (
 	IngressComponentName ComponentName = "IngressGateways"
 	EgressComponentName  ComponentName = "EgressGateways"
 
-	// Addon components
-	AddonComponentName ComponentName = "Addon"
+	// Addon root component
+	AddonComponentName ComponentName = "AddonComponents"
+
+	// Legacy addon components
```

Comment:
Why are these legacy? Because they used to be 'core' and need special treatment?